kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/tls/KeyPair.java

```diff
@@ -45,4 +45,20 @@ public SslContextBuilder forServer() {
                     e);
         }
     }
-}
+
+    @Override
+    public SslContextBuilder forClient() {
+        try {
+            return SslContextBuilder.forClient()
+                    .keyManager(new File(certificateFile), new File(privateKeyFile),
```

Comment:
What are the performance implications of calling `new File(....)` here? How often do we expect this path to be invoked? 

The one nice benefit of this approach is it does allow the certificate to be rotated seamlessly. However I think we can get that in other ways to avoid the need to load the file each time. 

I'm happy to pick up caching the certificate files in a separate PR once this is merged.

Reply:
Hi @SamBarker,

I didn't test performance for the moment, as I know the OS already cache files, performance optimization should be between java and os.

I'm not yet fluent with kroxylicious implementation, but if I remember right my test, connection and auth is only validated at the first request, it may be better to have a concret strategy about hat, like the oauth2 workflow for example, with an expiration validation time and a revoked certificate (not validated by the server). Si as you mentioned, the cache in fact.

Happy to contribute

Reply:
Once per connection sounds fine for now. We can revisit if profiling suggests the overheads are problematic.